### PR TITLE
[fix](year) fix `year()` results are not as expected

### LINKED ISSUE #13425: [Bug] `year()` results are not as expected
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

`year()` results are not as expected

### What You Expected?

```
MySQL [testdb]> select date, year(date) from test_year;
+------------+------------+
| date       | year(date) |
+------------+------------+
| 2021-01-01 |       2021 |
| 0000-01-01 |          0 |
| 9999-01-01 |       9999 |
+------------+------------+
```

### How to Reproduce?

```
CREATE TABLE `test_year` (
  `id` int null, 
  `date` date NULL
) ENGINE=OLAP
UNIQUE KEY(`id`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`id`) BUCKETS 2
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);

insert into test_year values(1, '2021-01-01'), (2, '0-01-01'),(3, '9999-01-01');

mysql> select date, year(date) from test_year;
+------------+--------------+
| date       | year(`date`) |
+------------+--------------+
| 9999-01-01 |         NULL |
| 2021-01-01 |         2021 |
| 0000-01-01 |         NULL |
+------------+--------------+
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #13425

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/date_time_transforms.h [hunk 1]
```diff
@@ -308,7 +308,7 @@ struct Transformer<FromType, ToType, ToYearImpl<FromType>> {
         }
 
         for (size_t i = 0; i < size; ++i) {
-            null_map_ptr[i] = to_ptr[i] <= MIN_YEAR || to_ptr[i] >= MAX_YEAR;
+            null_map_ptr[i] = to_ptr[i] < MIN_YEAR || to_ptr[i] > MAX_YEAR;
```
Comment (BiteTheDDDDt) [inline be/src/vec/functions/date_time_transforms.h:311]: year is unsigned type, we can remove year<0

Comment (luozenglin) [reply be/src/vec/functions/date_time_transforms.h]: done

File: be/src/vec/functions/date_time_transforms.h [hunk 2]
```diff
@@ -308,7 +308,7 @@ struct Transformer<FromType, ToType, ToYearImpl<FromType>> {
         }
 
         for (size_t i = 0; i < size; ++i) {
-            null_map_ptr[i] = to_ptr[i] <= MIN_YEAR || to_ptr[i] >= MAX_YEAR;
```
Comment (Gabriel39) [inline be/src/vec/functions/date_time_transforms.h:311]: This checking rule also should be same as FE. Could you add some cases for FE `year` function?

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
introduced by #13362

**Pull request merged**